flowmachine/flowmachine/features/location/redacted_labelled_spatial_aggregate.py

```diff
@@ -0,0 +1,62 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+from typing import List
+
+from flowmachine.features.location.labelled_spatial_aggregate import (
+    LabelledSpatialAggregate,
+)
+from flowmachine.core import Query
+from flowmachine.core.mixins import GeoDataMixin
+
+
+class RedactedLabelledSpatialAggregate(GeoDataMixin, Query):
+    """
+    Query that drops any locations that, when disaggregated by label, reveal a number of subscribers
+    less than redaction_threshold
+    Parameters
+    ----------
+    labelled_spatial_aggregate: LabelledSpatialAggregate
+        The LabelledSpatialAggregate query to redact
+    redaction_threshold: int default 15
+        If any labels within a location reveal less than this number of subscribers, that location is dropeed
+    """
+
+    def __init__(
+        self,
+        *,
+        labelled_spatial_aggregate: LabelledSpatialAggregate,
+        redaction_threshold: int = 15,
+    ):
+
+        self.labelled_spatial_aggregate = labelled_spatial_aggregate
+        self.redaction_threshold = redaction_threshold
+        super().__init__()
+
+    @property
+    def column_names(self) -> List[str]:
+        return self.labelled_spatial_aggregate.column_names
+
+    def _make_query(self):
```

Comment:
We discussed having this return the total (non-disaggregated) count for a location if any of the disaggregated counts are below the threshold, rather than returning nothing at all for that location. But actually, that's no different to using this approach (drop all rows for locations with any below-threshold counts) alongside a second, non-disaggregated, SpatialAggregate query

Reply:
Yes, agreed. It would also require a second redaction step to drop all locations where the total count is below the redaction threshold, which I think would end up being a bit messy.